packages/labs/compiler/src/test/source_map_test.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {SourceMapConsumer} from 'source-map';
+
+import {test} from 'uvu';
+// eslint-disable-next-line import/extensions
+import * as assert from 'uvu/assert';
+import {readFile} from 'fs/promises';
+import * as url from 'url';
+import * as path from 'path';
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+const TEST_FILES_DIR = path.resolve(
+  __dirname,
+  '../test_files/source_map_tests'
+);
+
+/**
+ * This helper asserts that a given position in the compiled file maps to a
+ * specific position in the authored file.
+ *
+ * To write these tests, use the editor to inspect line and column numbers in
+ * the source_map_tests directory.
+ *
+ * @param smc SourceMapConsumer
+ * @param generatedPosition The [line, column] for the generatedPosition.
+ * @param authoredPosition The [line, column] for the authoredPosition. May be
+ * null if there is no mapping.
+ */
+const assertSourceMapLocationMapping = ({
+  smc,
+  generatedPosition,
+  authoredPosition,
+  msg,
+}: {
+  smc: SourceMapConsumer;
+  generatedPosition: [number, number];
+  authoredPosition: [number | null, number | null];
+  msg: string;
+}) => {
+  const result = smc.originalPositionFor({
+    line: generatedPosition[0],
+    column: generatedPosition[1],
+  });
+
+  assert.equal([result.line, result.column], authoredPosition, msg);
+};
+
+const setupTest = async (filename: string): Promise<SourceMapConsumer> => {
+  const resolvedSourcemapPath = path.resolve(TEST_FILES_DIR, filename);
+  const map = await readFile(resolvedSourcemapPath, {encoding: 'utf-8'});
+  const smc = await new SourceMapConsumer(map);
+  return smc;
+};
+
+test('basic.ts', async () => {
+  const smc = await setupTest('basic.js.map');
```

Comment:
probably not a big deal with a single file test like this but maybe it's worth putting this in a `suiteSetup()` and using `suiteTeardown()` to call [`smc.destroy()`](https://github.com/mozilla/source-map#sourcemapconsumerprototypedestroy:~:text=When%20the%20SourceMapConsumer%20will%20no%20longer%20be%20used%20anymore%2C%20you%20must%20call%20its%20destroy%20method.)?

Reply:
Great catch. This is a small test so I think I'm inclined to add the `smc.destroy()` directly to the end of the test.

I tried adding a `test.after.each` which cleans up the smc, but noted that if the destructor throws outside the test, we get a 0/1 tests passed with an exit code of 0 – which makes it look like everything worked. Went with simpler approach which doesn't have this strange issue.

Reply:
since all the assertions are synchronous it's probably fine as is, though something about test code execution ordering makes me a bit uncomfortable.

how about something like this?
https://github.com/lit/lit/commit/c5169669f3dfd396f3544a739edf11d6169e7d4e

this seems to work on my end. is there an actual chance that `.destroy()` would fail? and should that then report as a test failure?
if so we can add try catch like this https://github.com/google/wireit/blob/df9c9e9d3dc4bd1c92933ef6d4e1919679390478/src/test/analysis.test.ts#L27-L34